src/software/ai/hl/stp/play/free_kick_play_test.py

```diff
@@ -134,125 +151,77 @@ def test_free_kick_play_enemy(simulated_test_runner, ball_pos, yellow_bot_pos):
-        tbots.Point(-3, -0.5),
-        tbots.Point(-3, -1.5),
-        tbots.Point(-3, -0.25),
-    ]
-
-    yellow_bots = [
-        tbots.Point(3, 0.25),
-        tbots.Point(3, 1.5),
-        tbots.Point(3, 0.5),
-        tbots.Point(3, -0.5),
-        tbots.Point(3, -1.5),
-        tbots.Point(3, -0.25),
-    ]
-
-    # Game Controller Setup
-    simulated_test_runner.gamecontroller.send_ci_input(
-        gc_command=Command.Type.STOP, team=Team.UNKNOWN
-    )
-    simulated_test_runner.gamecontroller.send_ci_input(
-        gc_command=Command.Type.FORCE_START, team=Team.UNKNOWN
-    )
-
-    # Force play override here
-    blue_play = Play()
-    blue_play.name = PlayName.FreeKickPlay
-
-    yellow_play = Play()
-    yellow_play.name = PlayName.EnemyFreekickPlay
-
-    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
-    simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(Play, yellow_play)
-
-    # Create world state
-    simulated_test_runner.simulator_proto_unix_io.send_proto(
-        WorldState,
-        create_world_state(
-            yellow_robot_locations=yellow_bots,
-            blue_robot_locations=blue_bots,
-            ball_location=ball_initial_pos,
-            ball_velocity=tbots.Vector(0, 0),
-        ),
-    )
-
-    # Always Validation
+def test_free_kick_play_both(simulated_test_runner, ball_initial_pos):
     # TODO- #2753 Validation
-    always_validation_sequence_set = [[]]
-
-    # Eventually Validation
-    # TODO- #2753 Validation
-    eventually_validation_sequence_set = [[]]
-
     simulated_test_runner.run_test(
-        eventually_validation_sequence_set=eventually_validation_sequence_set,
-        always_validation_sequence_set=always_validation_sequence_set,
+        setup=lambda test_setup_arg: free_kick_play_test_setup(
+            test_setup_arg, simulated_test_runner
+        ),
+        params=[
+            {
+                "blue_bots": [
+                    tbots.Point(-3, 0.25),
+                    tbots.Point(-3, 1.5),
+                    tbots.Point(-3, 0.5),
+                    tbots.Point(-3, -0.5),
+                    tbots.Point(-3, -1.5),
+                    tbots.Point(-3, -0.25),
+                ],
+                "yellow_bots": [
+                    tbots.Point(3, 0.25),
+                    tbots.Point(3, 1.5),
+                    tbots.Point(3, 0.5),
+                    tbots.Point(3, -0.5),
+                    tbots.Point(3, -1.5),
+                    tbots.Point(3, -0.25),
+                ],
+                "ball_initial_pos": ball_initial_pos,
+                "play_name": PlayName.EnemyFreekickPlay,
+            }
+        ],
+        inv_always_validation_sequence_set=[[]],
+        inv_eventually_validation_sequence_set=[[]],
+        ag_always_validation_sequence_set=[[]],
+        ag_eventually_validation_sequence_set=[[]],
         test_timeout_s=15,
     )
 
 
+free_kick_play_test_setup.__test__ = False
```

Comment:
What does this line do? Can you add a comment briefly explaining it (I think it's in other test files as well)